src/main/java/com/fasterxml/jackson/core/json/ByteSourceJsonBootstrapper.java

```diff
@@ -230,6 +230,12 @@ public Reader constructReader() throws IOException
                 InputStream in = _in;
 
                 if (in == null) {
+                    int length = _inputEnd - _inputPtr;
+                    if (length >= 0 && length <= 8192) {
```

Comment:
minor nit: It may not be worth checking `length >= 0`. It has a small perf impact and the length is very unlikely to negative and will just fail at the ByteArrayInputStream stage anyway. The length == 0 case will probably be more tidily handled with the StringReader than the ByteArrayInputStream route.

Reply:
I flipped the condition to short circuit for large inputs first `if (length <= STRING_READER_BYTE_ARRAY_LENGTH_LIMIT && length >= 0)`.

I'd prefer to keep the `length >= 0` as the array length check here should be negligible, and ensures the `InputStreamReader` continues handling any negative array length path.

Reply:
After thinking it through over night, I removed the `length >= 0` condition.